README.md

```diff
@@ -3,28 +3,33 @@
 [![GitHub Build Status](https://github.com/cisagov/ansible-role-manage-thp/workflows/build/badge.svg)](https://github.com/cisagov/ansible-role-manage-thp/actions)
 [![CodeQL](https://github.com/cisagov/ansible-role-manage-thp/workflows/CodeQL/badge.svg)](https://github.com/cisagov/ansible-role-manage-thp/actions/workflows/codeql-analysis.yml)
 
-This is a skeleton project that can be used to quickly get a new
-[cisagov](https://github.com/cisagov) Ansible role GitHub project
-started.  This skeleton project contains
-[licensing information](LICENSE), as well as
-[pre-commit hooks](https://pre-commit.com) and
-[GitHub Actions](https://github.com/features/actions) configurations
-appropriate for an Ansible role.
+An Ansible role that creates a SystemD unit to manage Transparent HugePage (THP)
+settings on Linux systems. This role is inspired by the
+[GekoCloud/ansible-role-disable-thp](https://github.com/GekoCloud/ansible-role-disable-thp)
+role.
 
 ## Requirements ##
 
 None.
 
 ## Role Variables ##
 
-None.
+> [!NOTE]
+> Although none of the variables that control settings are individually required,
+> at least one of them must be defined.
 
-<!--
 | Variable | Description | Default | Required |
 |----------|-------------|---------|----------|
-| optional_variable | Describe its purpose. | `default_value` | No |
-| required_variable | Describe its purpose. | n/a | Yes |
--->
+| manage\_thp\_defrag\_path | The sysfs path to control the Transparent HugePage `defrag` setting. | `/sys/kernel/mm/transparent_hugepage/defrag` | No |
+| manage\_thp\_defrag\_setting | If defined it must be a value of `always`, `defer`, `defer+madvise`, `madvise`, or `never`. Please see the [kernel documentation] for more information. | n/a | No |
+| manage\_thp\_enabled\_path | The sysfs path to control the Transparent HugePage `enabled` setting. | `/sys/kernel/mm/transparent_hugepage/enabled` | No |
+| manage\_thp\_enabled\_setting | A value of `always`, `madvise`, or `never`. Please see the [kernel documentation] for more information. | n/a | No |
+| manage\_thp\_service\_name | The name of the SystemD service that is created to manage Transparent HugePage settings. Please note that `.service` is appended to this value in the name of the unit file created. | `configure-transparent-hugepage-settings` | No |
+| manage\_thp\_shmem\_enabled\_path | The sysfs path to control the Transparent HugePage `shmem_enabled` setting. | `/sys/kernel/mm/transparent_hugepage/shmem_enabled` | No |
+| manage\_thp\_shmem\_enabled\_setting | If defined it must be a value of `always`, `within_size`, `advise`, `never`, `deny`, or `force`. Please see the [kernel documentation] for more information. | n/a | No |
```

Comment:
Alphabetize?
```suggestion
| manage\_thp\_shmem\_enabled\_setting | If defined it must be a value of `advise`, `always`, `deny`, `force`, `never`, or `within_size`. Please see the [kernel documentation] for more information. | n/a | No |
```

Reply:
The sorting for all of these is based on output like so:

```console
$ cat /sys/kernel/mm/transparent_hugepage/shmem_enabled
always within_size advise [never] deny force
```

I elected to duplicate the ordering to align with how it is presented if done manually.

Reply:
I'm fine with the ordering as it is.  I'm also fine if you want to alphabetize it.

Reply:
That's fine by me.  Can we put a comment somewhere in here so that we don't revisit this in the future.

Reply:
@dav3r @jsf9k would you take a look at [259b4a3](https://github.com/cisagov/ansible-role-manage-thp/pull/4/commits/259b4a32ff6962c2d6e2894687b377fb79e39774) and see what that does for you?

Reply:
That comment works for me, thanks!  I'm resolving this conversation.  @jsf9k feel free to reopen if you want any adjustments.

Reply:
Works for me too!